### PR TITLE
(CDAP-4117) Copied the fixed class from Twill.

### PR DESCRIPTION
**Pull request by chtyim**

### CODE REVIEW

**Comment by anew** [comment]
Is this because we would need another Twill release otherwise?

File: cdap-app-fabric/src/main/java/org/apache/twill/filesystem/FileContextLocationFactory.java [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.UnsupportedFileSystemException;
+
+import java.net.URI;
+import java.util.Objects;
+
+/**
+ * A {@link LocationFactory} implementation that uses {@link FileContext} to create {@link Location}.
+ *
+ * TODO: This class is copied from Twill for the fix in TWILL-163. Should be removed when upgraded to twill-0.8.0
```
Comment (sagarkapare) [inline cdap-app-fabric/src/main/java/org/apache/twill/filesystem/FileContextLocationFactory.java]: Should we have CDAP JIRA to track this?

**Comment by sagarkapare** [comment]
@chtyim one minor comment, otherwise LGTM :+1:

File: cdap-app-fabric/src/main/java/org/apache/twill/filesystem/FileContextLocationFactory.java (hunk 1 above)
Comment (anew) [inline cdap-app-fabric/src/main/java/org/apache/twill/filesystem/FileContextLocationFactory.java]: +1

**Comment by chtyim** [comment]
Yes. It's from twill-0.8.0, which hasn't release yet.

File: cdap-app-fabric/src/main/java/org/apache/twill/filesystem/FileContextLocationFactory.java (hunk 1 above)
Comment (chtyim) [inline cdap-app-fabric/src/main/java/org/apache/twill/filesystem/FileContextLocationFactory.java]: Added https://issues.cask.co/browse/CDAP-4923

**Comment by chtyim** [comment]
1 build passed on Travis and bamboo passed: http://builds.cask.co/browse/CDAP-DUT3555

There are some already known flaky test failure in Java8 in Travis.

**Pull request merged**